userApi/src/test/java/cs4337/group9/userapi/Service/UserServiceTest.java

```diff
@@ -0,0 +1,198 @@
+package cs4337.group9.userapi.Service;
+
+import cs4337.group9.mediumwebsite.Service.UserService;
+import cs4337.group9.mediumwebsite.Entity.UserEntity;
+import cs4337.group9.mediumwebsite.Entity.AdminActionEntity;
+import cs4337.group9.mediumwebsite.Exceptions.UserAlreadyExistsException;
+import cs4337.group9.mediumwebsite.Exceptions.UserNotFoundException;
+import cs4337.group9.mediumwebsite.Repostiory.UserRepository;
+import cs4337.group9.mediumwebsite.Repostiory.AdminActionRepository;
+import cs4337.group9.mediumwebsite.enums.Status;
+import cs4337.group9.mediumwebsite.enums.Role;
+import cs4337.group9.mediumwebsite.enums.Action;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.*;
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.Optional;
+import java.util.UUID;
+
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private AdminActionRepository adminActionRepository;
+
+    @InjectMocks
+    private UserService userService;
+
+    private UserEntity user;
+    private UUID userId;
+    private UUID adminId;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        userId = UUID.randomUUID();
+        adminId = UUID.randomUUID();
+        user = new UserEntity(userId, "testUser", "test@example.com", "password", Role.USER, Status.ACTIVE, "Test about", "profile.jpg", null, null);
+    }
+
+    @Test
+    void testCreateUser_UserAlreadyExists_ThrowsException() {
+        // Arrange
+        when(userRepository.findByEmail(user.getEmail())).thenReturn(Optional.of(user));
+
+        // Act & Assert
+        UserAlreadyExistsException exception = assertThrows(UserAlreadyExistsException.class, () -> {
+            userService.createUser(user);
+        });
+        assertEquals("User already exists with email: test@example.com", exception.getMessage());
```

Comment:
The `assertEquals()` message doesnt match the message acutally thrown by the exception 
`Expected :User already exists with email: test@example.com
Actual   :Account already created with the following email:  test@example.com`